webpack.config.js

```diff
@@ -0,0 +1,175 @@
+          {
+            loader: 'babel-loader',
+            query: {
+              cacheDirectory: true,
+              env: {
+                development: {
+                  presets: ['react-hmre'],
+                  plugins: [
+                    ['react-transform', {
+                      transforms: [{
+                        transform: 'react-transform-hmr',
+                        imports: ['react'],
+                        locals: ['module']
+                      }]
+                    }]
+                  ]
+                }
+              }
+            }
+          }
+        ]
+      },
+      {
+        test: /\.json$/,
+        exclude: /node_modules/,
+        loader: 'json-loader'
+      },
+      {
+        test: /\.css$/,
+        exclude: /node_modules/,
+        use: [
+          { loader: 'style-loader' },
+          { loader: 'css-loader' }
+        ]
+      }
+    ]
+  },
+  plugins: [
+    new FriendlyErrorsWebpackPlugin({
+      compilationSuccessInfo: {
+        messages: [`[Positron] Listening on http://localhost:${PORT} \n`]
+      }
+    }),
+    new ProgressBarPlugin(),
+    new WebpackNotifierPlugin(),
+    new webpack.DefinePlugin({
+      'process.env': {
+        'NODE_ENV': JSON.stringify(NODE_ENV)
+      }
+    }),
+    new webpack.NoEmitOnErrorsPlugin(),
+    new webpack.ProvidePlugin({
+      '$': 'jquery',
+      'jQuery': 'jquery',
+      'window.jQuery': 'jquery'
+    })
+  ],
+  resolve: {
+    extensions: ['.coffee', '.js', '.jsx', '.json'],
+    modules: [
+      'node_modules'
+    ]
+  }
+}
+
+// Development
+if (isDevelopment) {
+  config.plugins.push(new webpack.HotModuleReplacementPlugin())
+
+  // Staging
+} else if (isDeploy) {
+  config.devtool = '#source-map'
+
+  // Prod
+  if (isProduction) {
+    config.plugins.push(
+      new webpack.optimize.UglifyJsPlugin({
+        sourceMap: true
+      })
+    )
+  }
+}
+
+// Helpers
+
+function getEntrypoints () {
+  return findAssets('client/assets')
+}
+
+function findAssets (basePath) {
+  const files = fs.readdirSync(path.join(process.cwd(), basePath))
+
+  // Filter out .styl files
+  const validAssets = (file) => {
+    const whitelist = [ '.js', '.coffee' ]
+    const isValid = whitelist.some(extension => extension === path.extname(file))
+    return isValid
+  }
+
+  /**
+   * Construct key/value pairs representing Webpack entrypoints; e.g.,
+   * { desktop: [ path/to/subapp.asset.js ] }
+   */
+  const assets = files
+    .filter(validAssets)
+    .reduce((assetMap, file) => {
+      const fileName = path.basename(file, path.extname(file))
+      const asset = {
+        [fileName]: [
+          path.join(__dirname, basePath, file)
+        ]
+      }
+      if (isDevelopment) {
+        asset[fileName].unshift(
+          'webpack-hot-middleware/client?reload=true'
+        )
+      }
+
+      return {
+        ...assetMap,
+        ...asset
+      }
+    }, {})
+
+  return assets
+}
+
+module.exports = config
```

Comment:
Oh, Webpack and its long config files.

Reply:
LOL yeah -- especially when `$ ezel-assets` with all of its sensible defaults and extensibility via `.babelrc` is two words! The new webpack docs are unreal, too -- its intentionally powerful, but I have literally zero idea how they've brainstormed all of the possible usecases.